### DISCUSSION TITLE
Cache busting

### DISCUSSION THREAD

**Discussion by irmakcosarsahna** [vercel/next.js#30793] [Help]
First of all, I would like to point out that I use vercel.
When I make a new update after releasing the product, users do not receive the next version.
Users cannot open the site without clearing the Cache.
I was using service worker, turned it off and cleared all caches.

What I've done;
1. Use the build id.
```
 generateBuildId: async () => new Date().getTime()
```

2. I turned off the service worker.
```
  if ('serviceWorker' in navigator) {
            navigator.serviceWorker.getRegistrations().then(registrations => {
                (registrations || []).forEach((a, i) => {
                    a.unregister().then(bool => {});
                    if ((registrations || []).length - 1 === i) {
                        window.location.reload();
                    }
                })
            })
        }
```

3. I deleted all caches.
```
 self.addEventListener('activate', function(event) {
            event.waitUntil(
                caches.keys().then(function(cacheNames) {
                    return Promise.all(
                        cacheNames.map(function(cacheName) {
                            return caches.delete(cacheName);
                        })
                    );
                })
            );
        });
```

I want to learn the method of changing the hash here. Is there a method?

<img width="295" alt="Ekran Resmi 2021-11-02 13 03 20" src="https://user-images.githubusercontent.com/6817814/139827334-d2e930d5-23d2-4970-8b28-f96e7b3d2303.png">

How should I go about this? Can you help me ?

**Comment by Pieparker** [comment]
Simply put, the hash should be updated. I suspect that there may be something else at play (either the service worker wasn't de-registered or you're proxying your content via, say, Cloudflare). Do you have a URL/repro?
